OCPBUGS-17785: Document SR-IOV/FlowCollector config (#64448)
openshift/openshift-docs: modules/network-observability-SRIOV-configuration.adoc

// Module included in the following assemblies:
//
// network_observability/configuring-operator.adoc

:_content-type: PROCEDURE
[id="network-observability-SR-IOV-config_{context}"]
= Configure monitoring of SR-IOV interface traffic

> In order to collect traffic from a cluster with Single Root I/O Virtualization (SR-IOV) device, the `FlowCollector` `spec.agent.ebpf.privileged` specification value must be `true`. Then, the eBPF agent monitors the host network namespaces. When a pod with virtual functions (VF) interface is created, a new network namespace is created and the VF interface is migrated from the host network namespace to the pod network namespace.

Reviewer: the need for `privileged` to be able to detect none host netns, today ebpf agent by default watch host netns 

also the VF migrated under certain configuration in sriov policy where we have to specify `IPAM` without no migration will take place, @memodi can share more configs here as needed

Author: Discussed with @msherif1234 on Slack and the following updates will be made based on this comment:
- Clarify that egardless to the privileged setting host netns is always visible AND to see other netns we have to set privileged to true
-  make it clear VF interfaces won't always migrate from host netns to pod netns; it only migrates under certain configurations. Add for example if SRIOV policy uses IPAM.

Author: Addressed. I added the `SRIOVNetwork` CR IPAM config to the `Prerequisites`. Is that OK @msherif1234 ?

// Module included in the following assemblies:
//
// network_observability/configuring-operator.adoc

:_content-type: PROCEDURE
[id="network-observability-SR-IOV-config_{context}"]

> = Configuring the monitoring of SR-IOV interface traffic

Reviewer 2: Title is fine as is, but adding an alternative suggestion of "Configuring monitoring for SR-IOV interface traffic"

Author: Your suggestion is much cleaner, thank you!

In order to collect traffic from a cluster with Single Root I/O Virtualization (SR-IOV) device, the `FlowCollector` `spec.agent.ebpf.privileged` specification value must be `true`. Then, the eBPF agent monitors other network namespaces in addition to the host network namespaces, which are monitored by default. When a pod with a virtual functions (VF) interface is created, a new network namespace is created. The VF interface is migrated from the host network namespace to the pod network namespace.

> 

Reviewer: that isn't always the case though when using special sriov policy config that can happen also cc'd @bn222 for his comments

Author: Ok...I updated this to say "With `SRIOVNetwork` policy `IPAM` configurations specified, the VF interface is migrated from the host network namespace to the pod network namespace." Does that help? Or could be clearer?

> In order to collect traffic from a cluster with Single Root I/O Virtualization (SR-IOV) device, the `FlowCollector` `spec.agent.ebpf.privileged` specification value must be `true`. Then, the eBPF agent monitors other network namespaces in addition to the host network namespaces, which are monitored by default. When a pod with a virtual functions (VF) interface is created, a new network namespace is created. With `SRIOVNetwork` policy `IPAM` configurations specified, the VF interface is migrated from the host network namespace to the pod network namespace.

Reviewer 2: * Should this have "a" in front"?
* I know people call these different things, but I've never seen "specification value" used. Maybe just say "field"?
* I'm not sure if starting the second sentence with "Then" makes sense, because you didn't tell them an action to take, you just made a statement saying that it has to be true. Maybe if you change the first sentence to say "you must set the .... to `true`." it would make better sense? And it would be more active, so that might be the way to go, if it makes sense. If it doesn't make sense (because maybe it's not this person that set it?) then maybe reconsider replacing "Then" with a different phrase.
* Should virtual function be singular? I see it both ways in the repo

```suggestion
In order to collect traffic from a cluster with a Single Root I/O Virtualization (SR-IOV) device, you must set the `FlowCollector` `spec.agent.ebpf.privileged` field value to `true`. Then, the eBPF agent monitors other network namespaces in addition to the host network namespaces, which are monitored by default. When a pod with a virtual function (VF) interface is created, a new network namespace is created. With `SRIOVNetwork` policy `IPAM` configurations specified, the VF interface is migrated from the host network namespace to the pod network namespace.
```

Author: - I do think I'm missing an article here, good catch. 
- People really do call these different things. I think field is good. I think someone has recommended that I specifically say `value` in another peer review but I don't know for sure. 
- I think changing to "you must set the ... to `true`" is a good suggestion, and I can definitely see how it makes more sense with the "Then".
- I think in this case it is singular.

> 

Reviewer 3: > the VF interface is migrated from the host network namespace to the pod network namespace.

is this technical detail required in docs?

Author: @memodi I think @msherif1234 was wanting to make really clear that the 'SRIOVNetwork` spec is crucial to the migration. I'll let him weigh in and see if maybe we should tone down the technical details here.

Reviewer 3: Got it, I understand it better now, we can that info ON. Resolving this.

> .Prerequisite

Reviewer 2: Should be plural

```suggestion
.Prerequisites
```

> * Access to an {product-title} cluster with SR-IOV device. 

Reviewer 2: Same question on if there should be an "a" in front of "SR-IOV device"?

> * The `SRIOVNetwork` custom resource (CR) specifies `IPAM` configured with an IP address from the range in which interface has the IP or from other plugins.

Reviewer 2: I'm unsure what this is saying, is there a "the" missing in front of "interface"? Also the "in which" might be throwing me off. And maybe I don't understand what the "or from other plugins" refers to. Like what's the alternative to other plugins?

I know this isn't my area of expertise so I can't give a suggestion, but I think it's hard to follow what you mean here, so maybe worth another look

Author: Yeah this could definitely be clearer, you're right. This was written in a crunch, and it definitely shows. I think this may be more like it:
The `SRIOVNetwork` custom resource (CR) `spec.ipam` configuration must be set with an IP address from the range that the interface lists or from other plugins.

.Procedure
. In the web console, navigate to *Operators* -> *Installed Operators*.
. Under the *Provided APIs* heading for the *NetObserv Operator*, select *Flow Collector*. 

> . Select *cluster* then select the *YAML* tab.

Reviewer 2: ```suggestion
. Select *cluster* and then select the *YAML* tab.
```

. Configure the `FlowCollector` custom resource. A sample configuration is as follows:
+
[id="network-observability-flowcollector-configuring-SRIOV-monitoring{context}"]
.Configure `FlowCollector` for packet drops
[source,yaml]
----
apiVersion: flows.netobserv.io/v1alpha1
kind: FlowCollector
metadata:
  name: cluster
spec:
  namespace: netobserv
  deploymentModel: DIRECT
  agent:
    type: EBPF
    ebpf:        

>       privileged: true             <1> 

Reviewer 2: Should the <1> be just a single space after "true" here? I know that I sometimes put mine all right aligned when I have multiple, but this is just a single one, and it isn't right aligned past the longest line (line 24)

Author: Yeah I clearly got really happy with my space bar 😆 Could definitely have less spacing.

----

> <1> Note that the `spec.agent.ebpf.privileged` specification value must be `true` for packet drop tracking to be enabled.

Reviewer 2: * I don't think the "Note that" is necessary
* Same comment about specification value

```suggestion
<1> The `spec.agent.ebpf.privileged` field value must be `true` to enable packet drop tracking.
```
